lib/Backend/GlobOpt.cpp

```diff
@@ -2644,7 +2644,7 @@ GlobOpt::OptInstr(IR::Instr *&instr, bool* isInstrRemoved)
         this->InsertByteCodeUses(instr);
     }
 
-    if (!IsLoopPrePass() && instr->HasBailOutInfo())
```

Comment:
Out of curiosity, what is different about BailForDebugger?

Reply:
It is unique in the sense that while its inserted after an instruction, the offset it bails out to is still that of the instruction it was inserted after (different from post-op bailouts which bail out to the next offset). This presents a problem in code like:

     ByteCodeUses T1
     T2 = LdFld ###
     BailForDebugger
     .
     .
         = T2

if  `T1` and `T2` correspond to the same bytecode register, moving the `ByteCodeUses` next to `BailForDebugger` will make both `T1` and `T2` live upwards at the same time and sharing the same byte code reg.